src/main/java/one/eim/gravitycontrol/Compatibility.java

```diff
@@ -0,0 +1,46 @@
+package one.eim.gravitycontrol;
+
+import org.bukkit.Bukkit;
+import org.bukkit.plugin.Plugin;
+import java.util.function.Consumer;
+
+public class Compatibility {
+
+    public static <T> T callMethod(Class<?> clazz, Object object, String methodName, Class<?>[] parameterTypes, Object... args) {
+        try {
+            return (T) clazz.getDeclaredMethod(methodName, parameterTypes).invoke(object, args);
+        } catch (Throwable t) {
+            throw new IllegalStateException(t);
+        }
+    }
+
+    public static <T> T callMethod(Object object, String methodName, Class<?>[] parameterTypes, Object... args) {
+        return callMethod(object.getClass(), object, methodName, parameterTypes, args);
+    }
+
+    public static <T> T callMethod(Class<?> clazz, String methodName) {
+        return callMethod(clazz, null, methodName, new Class[]{});
+    }
+
+    private static boolean methodExist(Class<?> clazz, String methodName, Class<?>... parameterTypes) {
+        try {
+            clazz.getDeclaredMethod(methodName, parameterTypes);
+            return true;
+        } catch (Throwable ignored) {}
+        return false;
+    }
+
+    public static boolean isFolia() {
+        return methodExist(Bukkit.class, "getGlobalRegionScheduler");
+    }
```

Comment:
all of this reflection—bar checking if the method exists—shouldn't be necessary. Just

```java
if (IS_FOLIA) {
  Bukkit.getAsyncScheduler().runNow(...)
} else {
  Bukkit.getScheduler().runTaskAsynchronously(...)
}
```

Additionally, I would cache if it is folia by moving the methodExists check into a static block.